components/Link/index.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { Popover } from '@wordpress/components';
+import { isKeyboardEvent, ENTER } from '@wordpress/keycodes';
+import { placeCaretAtHorizontalEdge } from '@wordpress/dom';
+import {
+	__experimentalLinkControl as LinkControl,
+	useBlockProps,
+	RichText,
+} from '@wordpress/block-editor';
+import { isURL, prependHTTP, safeDecodeURI } from '@wordpress/url';
+
+/**
+ * Removes HTML from a given string.
+ * Note the does not provide XSS protection or otherwise attempt
+ * to filter strings with malicious intent.
+ *
+ * See also: https://github.com/WordPress/gutenberg/pull/35539
+ *
+ * @param {string} html the string from which HTML should be removed.
+ * @returns {string} the "cleaned" string.
+ */
+function navStripHTML(html) {
+	const doc = document.implementation.createHTMLDocument('');
+	doc.body.innerHTML = html;
+	return doc.body.textContent || '';
+}
+
+/**
+ * Given the Link block's type attribute, return the query params to give to
+ * /wp/v2/search.
+ *
+ * @param {string} type Link block's type attribute.
+ * @param {string} kind Link block's entity of kind (post-type|taxonomy)
+ * @returns {{ type?: string, subtype?: string }} Search query params.
+ */
+function getSuggestionsQuery(type, kind) {
+	switch (type) {
+		case 'post':
+		case 'page':
+			return { type: 'post', subtype: type };
+		case 'category':
+			return { type: 'term', subtype: 'category' };
+		case 'tag':
+			return { type: 'term', subtype: 'post_tag' };
+		case 'post_format':
+			return { type: 'post-format' };
+		default:
+			if (kind === 'taxonomy') {
+				return { type: 'term', subtype: type };
+			}
+			if (kind === 'post-type') {
+				return { type: 'post', subtype: type };
+			}
+			return {};
+	}
+}
+
+/**
+ * Link component that can be used inside other Gutenberg blocks for setting up URLs.
+ *
+ * The link should not be visible if the block is not focused. This will mainain nicer
+ * visuals in the block editor as a whole.
+ *
+ * @param props {object}
+ * @param props.attributes {object}
+ * @param props.setAttributes {function}
+ *
+ * @returns JSX.Element
+ */
+const Link = ({
+	label,
+	type,
+	opensInNewTab,
+	url,
+	onLinkChange,
+	onTextChange,
+	kind,
+	placeholder,
+	isSelected,
```

Comment:
Instead of relying on the developer to pass the `isSelected` prop here we can use the `useBlockEditContext` hook to get access to the current block from within our component.

```js
import { 	useBlockEditContext } from '@wordpress/block-editor';

const { isSelected } = useBlockEditContext();
```

Reply:
Thanks! This was updated